admin/admin-functions.php

```diff
@@ -0,0 +1,139 @@
+	echo '<label for="poem_date">Date: </label>';
+	echo '<input type="text" id="poem_date" name="poem_date" class="poetwp-datepicker" value="' . esc_attr( $date ) . '" size="25" />';
+	echo '<label for="poem_notes">Notes: </label>';
+	wp_editor(
+		$notes,
+		'poem_notes',
+		array(
+			'textarea_name' => 'poem_notes',
+			'media_buttons' => false,
+			'textarea_rows' => 5,
+			'teeny'         => true,
+		)
+	);
+}
+
+/**
+ * Save the meta box data
+ *
+ * @param int $post_id The ID of the post being saved.
+ */
+function poetwp_save_poem_details( $post_id ) {
+	// Check if our nonce is set and verify it.
+	if ( ! isset( $_POST['poetwp_poem_details_nonce'] ) || ! wp_verify_nonce( $_POST['poetwp_poem_details_nonce'], 'poetwp_save_poem_details' ) ) {
+		return;
+	}
+
+	// Check if this is an autosave.
+	if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+		return;
+	}
+
+	// Check the user's permissions.
+	if ( ! current_user_can( 'edit_post', $post_id ) ) {
+		return;
+	}
+
+	// Sanitize and save the meta data.
+	if ( isset( $_POST['poem_date'] ) ) {
+		update_post_meta( $post_id, 'poem_date', sanitize_text_field( $_POST['poem_date'] ) );
+	}
+	if ( isset( $_POST['poem_notes'] ) ) {
+		update_post_meta( $post_id, 'poem_notes', wp_kses_post( $_POST['poem_notes'] ) );
+	}
+}
+// Hook the save function to the 'save_post_poem' action.
+add_action( 'save_post_poem', 'poetwp_save_poem_details' );
+
+// Add action to enqueue datepicker scripts and styles.
+add_action( 'admin_enqueue_scripts', 'poetwp_enqueue_admin_assets' );
+
+/**
+ * Enqueue admin scripts and styles.
+ *
+ * @return void
+ */
+function poetwp_enqueue_admin_assets() {
+	// Enqueue jQuery UI Datepicker script and styles.
+	wp_enqueue_script( 'jquery-ui-datepicker' );
+	wp_enqueue_style( 'jquery-ui-css', '//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css', array(), '1.12.1' );
+
+	// Enqueue the custom admin stylesheet.
+	wp_enqueue_style( 'poetwp-admin-styles', plugin_dir_url( __FILE__ ) . 'css/poetwp-admin.css', array(), '1.0.0' );
+}
+
+// Add action to enqueue admin assets.
+add_action( 'admin_enqueue_scripts', 'poetwp_enqueue_admin_assets' );
+
+/**
+ * Enqueue block editor assets.
+ *
+ * @return void
+ */
+function poetwp_enqueue_block_editor_assets() {
+	wp_enqueue_script(
+		'poetwp-admin-script',
+		plugin_dir_url( __FILE__ ) . 'js/poetwp-admin.js',
+		array( 'wp-blocks', 'wp-dom-ready', 'wp-edit-post' ),
+		filemtime( plugin_dir_path( __FILE__ ) . 'js/poetwp-admin.js' ),
+		true
+	);
+}
+
+// Add action to enqueue block editor assets.
+add_action( 'enqueue_block_editor_assets', 'poetwp_enqueue_block_editor_assets' );
+
+/**
+ * Initialize the datepicker on the poem date field.
+ *
+ * @return void
+ */
+function poetwp_initialize_datepicker() {
+	?>
+	<script type="text/javascript">
+	jQuery(document).ready(function($) {
+		$('#poem_date').datepicker({
+			dateFormat: 'd MM yy',
+			changeMonth: true,
+			changeYear: true
+		});
+	});
+	</script>
+	<?php
+}
```

Comment:
Rather than adding a block of JS, consider using `wp_add_inline_script()` function call that references an enqueued script, so that it only gets printed out if the script was indeed output.

You might also like to avoid hard-coding a dateFormat and use the date format already stored in the WP settings for consistency.

Reply:
Noted #23